src/helper/CMakeLists.txt

```diff
@@ -32,6 +34,18 @@ endif()
 
 add_executable(sddm-helper ${HELPER_SOURCES})
 target_link_libraries(sddm-helper Qt5::Network Qt5::DBus Qt5::Qml)
+if(${CMAKE_SYSTEM} MATCHES "FreeBSD")
```

Comment:
CMAKE_SYSTEM is ${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_VERSION}, or ${CMAKE_SYSTEM_NAME} if CMAKE_SYSTEM_VERSION is empty which means that this might not always work.
See https://cmake.org/cmake/help/v3.4/variable/CMAKE_SYSTEM.html

Use CMAKE_SYSTEM_NAME instead.